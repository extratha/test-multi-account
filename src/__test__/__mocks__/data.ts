export const data = {
  test: 'test'
}