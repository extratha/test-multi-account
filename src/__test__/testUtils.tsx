import { RenderOptions, RenderResult, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SWRConfig } from 'swr';
import { theme } from '@/config/config-mui/theme';
import { ThemeProvider as MUIThemeProvider } from '@mui/material/styles';
import { NextIntlClientProvider } from 'next-intl';
import commonTh from '../../public/locales/th/common.json';
import commonEn from '../../public/locales/th/common.json';
type RenderWithProvider = (
  elm: React.ReactElement,
  renderOptions?: ProviderOptions,
) => RenderResult;

export type ProviderOptions = {
  formOptions?: Record<string, unknown>;
} & RenderOptions;

const messages = {
  ...commonTh,
  ...commonEn,
};

const renderWithProviders: RenderWithProvider = (
  component,
  renderOptions?,
  locale = 'th',
) => {
  const wrapper = ({ children }: { children: React.ReactNode }) => (
    <NextIntlClientProvider locale={locale} messages={messages}>
      <MUIThemeProvider theme={theme}>
        <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      </MUIThemeProvider>
    </NextIntlClientProvider>
  );

  return render(component, {
    wrapper,
    ...renderOptions,
  });
};

export { renderWithProviders as render, userEvent };
export * from '@testing-library/react';
