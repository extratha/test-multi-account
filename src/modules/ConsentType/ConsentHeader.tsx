import { alpha, AppBar, styled, Toolbar, Typography } from "@mui/material";

export interface ConsentHeaderProps {
  title: string;
}

const HeaderBar = styled(AppBar)(({ theme }) => ({
  color: theme.palette.text.hight,
  backgroundColor: theme.palette.common.white,
  boxShadow: `0px 4px 6px -1px ${alpha(theme.palette.common.black, 0.1)}`,
  "& > .MuiToolbar-root": {
    justifyContent: "center",
  },
}));

const ConsentHeader = ({ title }: ConsentHeaderProps) => {
  return (
    <HeaderBar>
      <Toolbar variant="dense">
        <Typography variant="bodyBold" textAlign="center">
          {title}
        </Typography>
      </Toolbar>
    </HeaderBar>
  );
};

export default ConsentHeader;
