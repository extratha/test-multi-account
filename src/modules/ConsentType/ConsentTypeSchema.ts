import useTranslation from "@/locales/useLocale";
import * as yup from "yup";

const useSubmitConsentSchema = () => {
  const { translation } = useTranslation();

  return yup.object().shape({
    agreement: yup.boolean().required(translation("Common.validation.require")),
  });
};

export default useSubmitConsentSchema;
