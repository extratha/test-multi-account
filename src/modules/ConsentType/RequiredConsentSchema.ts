import * as yup from "yup";

import useTranslation from "@/locales/useLocale";

// TODO: unit test
const useConsentTypeSchema = () => {
  const { translation } = useTranslation();

  return yup.object().shape({
    agreement: yup
      .boolean()
      .required(translation("Common.validation.require"))
      .test("agreement", translation("Common.validation.require"), (value) => {
        return value;
      }),
  });
};

export default useConsentTypeSchema;
