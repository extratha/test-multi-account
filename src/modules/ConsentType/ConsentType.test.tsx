import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import MockAdapter from "axios-mock-adapter";

import { mockPrivacyPolicyData, mockTermsAndConsData } from "@/__mocks__/data";
import * as Api from "@/api/api";
import { apiAxios } from "@/api/api";
import { CONSENT_TYPE } from "@/constant";
import { spyUseParams, spyUseRouter, SpyUseRouter } from "@/testUtils/navigation";
import { API, flushPromise, render } from "@/testUtils/testUtils";
import ConsentType, { ConsentTypeParams } from ".";

describe("ConsentType", () => {
  let mockApiAdapter: MockAdapter;
  let spyRouter: SpyUseRouter;
  let params: ConsentTypeParams;

  beforeEach(() => {
    params = { consentType: CONSENT_TYPE.TERMS_CONDITIONS };

    spyUseParams(params);
    spyRouter = spyUseRouter();

    jest.spyOn(Api, "getConsent");
    jest.spyOn(Api, "submitConsent");

    mockApiAdapter = new MockAdapter(apiAxios);
    mockApiAdapter.onGet(API.TERMS_AND_CONDITIONS).reply(200, mockTermsAndConsData);
    mockApiAdapter.onGet(API.PRIVACY_POLICY).reply(200, mockPrivacyPolicyData);
    mockApiAdapter.onPost(API.CONSENT).reply(201);
  });

  const renderConsentType = async () => {
    const view = render(<ConsentType />);
    await flushPromise();
    return view;
  };

  it("should render correctly", async () => {
    const { asFragment } = await renderConsentType();
    expect(asFragment()).toMatchSnapshot();
  });

  it("should not display consent when fetch consent error", async () => {
    mockApiAdapter.onGet(API.PRIVACY_POLICY).networkError();

    await renderConsentType();

    expect(screen.queryByTestId("privacy-policy-consent")).not.toBeInTheDocument();
  });

  it("should not navigate to Home page when submit consent error", async () => {
    mockApiAdapter.onPost(API.CONSENT).networkError();

    await renderConsentType();

    await userEvent.click(screen.getByTestId("agreement-checkbox"));
    await userEvent.click(screen.getByTestId("submit-button"));

    expect(spyRouter.push).not.toHaveBeenCalled();
  });
});
