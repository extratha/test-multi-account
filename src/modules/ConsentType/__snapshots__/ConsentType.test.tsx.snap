// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsentType should render correctly 1`] = `
<DocumentFragment>
  <title>
    ข้อกำหนดและเงื่อนไข
  </title>
  .emotion-0 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
  background-color: #eceff1;
}

<div
    class="MuiStack-root emotion-0"
  />
  .emotion-0 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  padding-left: 16px;
  padding-right: 16px;
  max-width: 1024px;
  margin: 0px auto;
  padding: 24px 16px;
  padding-top: 72px;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

.emotion-1 {
  background-color: #fff;
  color: unset;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 16px;
  box-shadow: none;
  padding: 36px 40px;
}

.emotion-2 {
  background-color: #FFFFFF;
  color: #212121;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  box-shadow: 0px 4px 6px -1px rgba(0, 0, 0, 0.1);
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  box-sizing: border-box;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: fixed;
  z-index: 1100;
  top: 0;
  left: auto;
  right: 0;
}

@media print {
  .emotion-2 {
    position: absolute;
  }
}

.emotion-2>.MuiToolbar-root {
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-3 {
  position: relative;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
  min-height: 48px;
}

@media (min-width:600px) {
  .emotion-3 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

.emotion-4 {
  margin: 0;
  font-size: 16px;
  font-weight: 700;
  line-height: 1.5;
  text-align: center;
}

.emotion-5 {
  margin: 0;
  font-size: 20px;
  font-weight: 700;
  line-height: 28px;
}

.emotion-6 {
  margin: 8px 0px 24px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0;
  border-style: solid;
  border-color: rgba(0, 0, 0, 0.12);
  border-bottom-width: thin;
}

.emotion-7 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-7>:not(style):not(style) {
  margin: 0;
}

.emotion-7>:not(style)~:not(style) {
  margin-top: 24px;
}

.emotion-8 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
  border-radius: 12px;
  background-color: #FAFAFA;
}

.emotion-9 {
  font-size: 16px;
  line-height: 28px;
}

.emotion-9>h3:first-of-type,
.emotion-9>p:first-of-type {
  margin-top: 0px;
}

.emotion-9>h3:last-of-type,
.emotion-9>p:last-of-type {
  margin-bottom: 0px;
}

.emotion-9 ul {
  padding-left: 24px;
  list-style-type: disc;
}

.emotion-9 ol {
  list-style-type: none;
  counter-reset: item;
  padding: 0px;
}

.emotion-9 ol>li {
  display: table-row;
  counter-increment: item;
  border-collapse: separate;
  border-spacing: 0px 6px;
}

.emotion-9 ol>li:before {
  display: table-cell;
  content: counters(item, ".") ". ";
  padding-right: 6px;
  white-space: nowrap;
}

.emotion-9 ol>ol,
.emotion-9 ol>ul {
  padding-left: 24px;
}

.emotion-9 ol>ol>li>ol,
.emotion-9 ol>ul>li>ol,
.emotion-9 ol>ol>li>ul,
.emotion-9 ol>ul>li>ul {
  padding: 0px;
}

.emotion-9 li {
  margin: 16px 0px;
}

.emotion-9 li ol>li {
  border-collapse: separate;
  border-spacing: 0px 6px;
}

.emotion-9 li ol>li:before {
  content: counters(item, ".") " ";
  padding-right: 6px;
}

.emotion-9 table {
  width: 100%;
  border-collapse: collapse;
  table-layout: fixed;
  margin: 16px 0px;
}

.emotion-9 table li {
  margin: 0px;
}

.emotion-9 th {
  height: 48px;
  text-align: center;
  color: #ffffff;
  border: 1px solid #000000;
  border-right-color: #ffffff;
  background-color: #002060;
}

.emotion-9 th:last-child {
  border-color: #000000;
}

.emotion-9 td {
  border: 1px solid #000000;
  padding: 8px;
  vertical-align: top;
}

.emotion-11 {
  margin: 0;
  font-size: 18px;
  font-weight: 700;
  line-height: 24px;
  margin-bottom: 24px;
}

.emotion-12 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #cfd8dc;
  background-color: #fff;
}

.emotion-12:first-of-type {
  border-radius: 12px 12px 0 0;
}

.emotion-12:last-of-type {
  border-radius: 0 0 12px 12px;
}

.emotion-13 {
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  cursor: pointer;
  padding: 24px;
}

.emotion-15 {
  max-width: 52px;
  max-height: 52px;
}

.emotion-16 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-left: 12px;
}

.emotion-16>:not(style):not(style) {
  margin: 0;
}

.emotion-16>:not(style)~:not(style) {
  margin-top: 4px;
}

.emotion-17 {
  margin: 0;
  font-size: 16px;
  font-weight: 500;
  line-height: 1.5;
}

.emotion-18 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-19 {
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 50%;
  padding: 8px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: rgba(0, 0, 0, 0.54);
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  overflow: visible;
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-19::-moz-focus-inner {
  border-style: none;
}

.emotion-19.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-19 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-19:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-19:hover {
    background-color: transparent;
  }
}

.emotion-19.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-20 {
  width: 24px;
  height: 24px;
}

.emotion-21 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-22 {
  height: 0;
  overflow: hidden;
  -webkit-transition: height 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: height 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  visibility: hidden;
}

.emotion-23 {
  display: flex;
  width: 100%;
}

.emotion-24 {
  width: 100%;
}

.emotion-25 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 16px;
}

.emotion-72 {
  margin-top: 24px;
}

.emotion-73 {
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  margin-left: -11px;
  margin-right: 16px;
}

.emotion-73.Mui-disabled {
  cursor: default;
}

.emotion-73 .MuiFormControlLabel-label.Mui-disabled {
  color: #B0BEC5;
}

.emotion-73>.MuiCheckbox-root {
  color: #cfd8dc;
}

.emotion-73>.Mui-checked {
  color: #2196F3;
}

.emotion-73>.MuiTypography-root {
  font-weight: 500;
  color: #78909c;
}

.emotion-74 {
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 50%;
  padding: 9px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: rgba(0, 0, 0, 0.6);
}

.emotion-74::-moz-focus-inner {
  border-style: none;
}

.emotion-74.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-74 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-74:hover {
  background-color: rgba(33, 150, 243, 0.04);
}

@media (hover: none) {
  .emotion-74:hover {
    background-color: transparent;
  }
}

.emotion-74.Mui-checked,
.emotion-74.MuiCheckbox-indeterminate {
  color: #2196F3;
}

.emotion-74.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-75 {
  cursor: inherit;
  position: absolute;
  opacity: 0;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.emotion-76 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-78 {
  margin: 0;
  font-family: fontFamily;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
}

.emotion-79 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 16px 0px;
}

.emotion-80 {
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: #2196F3;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 8px;
  padding: 6px 16px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
  font-family: fontFamily;
  font-weight: 700;
  font-size: 14px;
  line-height: 1.5;
  text-transform: uppercase;
  min-width: 64px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  height: 44px;
  width: 100%;
  max-width: 400px;
}

.emotion-80::-moz-focus-inner {
  border-style: none;
}

.emotion-80.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-80 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-80:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(23, 105, 170);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-80:hover {
    background-color: #2196F3;
  }
}

.emotion-80:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-80.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-80.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-80.Mui-disabled {
  background-color: #FAFAFA;
  colors: #B0BEC5;
}

<div
    class="MuiContainer-root MuiContainer-maxWidthLg emotion-0"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 emotion-1"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed emotion-2"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense emotion-3"
        >
          <span
            class="MuiTypography-root MuiTypography-bodyBold emotion-4"
          >
            ข้อกำหนดและเงื่อนไข
          </span>
        </div>
      </header>
      <span
        class="MuiTypography-root MuiTypography-titleLargeBold emotion-5"
      >
        ข้อกำหนดและเงื่อนไข
      </span>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth emotion-6"
      />
      <div
        class="MuiStack-root emotion-7"
      >
        <div
          class="MuiStack-root emotion-8"
        >
          <div
            class="emotion-9"
            data-testid="consent-type-content"
          >
            **Terms of Service**

Welcome to our service. By using our service, you agree to the following terms...

**Privacy Policy**

We value your privacy. This policy explains how we collect and use your information...
          </div>
        </div>
        <div
          class="MuiStack-root emotion-8"
          data-testid="consent-type-service"
        >
          <span
            class="MuiTypography-root MuiTypography-titleBold emotion-11"
          >
            All Services
          </span>
          <div
            class="MuiStack-root emotion-12"
          >
            <div
              class="MuiStack-root emotion-13"
              data-testid="consent-type-service-0"
            >
              <div
                class="MuiBox-root emotion-14"
              >
                <img
                  alt=""
                  class="emotion-15"
                  data-testid="consent-type-service-0-logo"
                  src="https://cdn-dev-playground.cariva.co.th/terms/images/logo.png"
                />
              </div>
              <div
                class="MuiStack-root emotion-16"
              >
                <span
                  class="MuiTypography-root MuiTypography-bodyMedium emotion-17"
                  data-testid="consent-type-service-0-title"
                >
                  mock-test-title
                </span>
                <svg
                  data-testid="consent-type-service-0-detail-icon"
                />
              </div>
              <div
                class="MuiStack-root emotion-18"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-19"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    class="emotion-20"
                  />
                  <span
                    class="MuiTouchRipple-root emotion-21"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-22"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper MuiCollapse-vertical emotion-23"
              >
                <div
                  class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-24"
                >
                  <div
                    class="MuiStack-root emotion-25"
                    data-testid="consent-type-service-0-content"
                  >
                    <div
                      class="emotion-9"
                      data-testid="consent-type-content"
                    >
                      **Service Overview**

AI Interpret provides advanced language translation capabilities...

**Features**

- Real-time translation
- Multi-language support
- High accuracy rates

**Usage Instructions**

1. Select your input and output languages.
2. Speak or type your text.
3. Receive the translated output.
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiStack-root emotion-12"
          >
            <div
              class="MuiStack-root emotion-13"
              data-testid="consent-type-service-1"
            >
              <div
                class="MuiBox-root emotion-14"
              >
                <img
                  alt=""
                  class="emotion-15"
                  data-testid="consent-type-service-1-logo"
                  src="https://cdn-dev-playground.cariva.co.th/terms/images/logo.png"
                />
              </div>
              <div
                class="MuiStack-root emotion-16"
              >
                <span
                  class="MuiTypography-root MuiTypography-bodyMedium emotion-17"
                  data-testid="consent-type-service-1-title"
                >
                  Health Visualization
                </span>
                <svg
                  data-testid="consent-type-service-1-detail-icon"
                />
              </div>
              <div
                class="MuiStack-root emotion-18"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-19"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    class="emotion-20"
                  />
                  <span
                    class="MuiTouchRipple-root emotion-21"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-22"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper MuiCollapse-vertical emotion-23"
              >
                <div
                  class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-24"
                >
                  <div
                    class="MuiStack-root emotion-25"
                    data-testid="consent-type-service-1-content"
                  >
                    <div
                      class="emotion-9"
                      data-testid="consent-type-content"
                    >
                      **Service Overview**

Health Visualization helps you understand your health data through visual representations...

**Features**

- Detailed health metrics
- Trend analysis
- Personalized insights

**Usage Instructions**

1. Upload your health data.
2. Choose the metrics you want to visualize.
3. View the visualized data and insights.
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiStack-root emotion-12"
          >
            <div
              class="MuiStack-root emotion-13"
              data-testid="consent-type-service-2"
            >
              <div
                class="MuiBox-root emotion-14"
              >
                <img
                  alt=""
                  class="emotion-15"
                  data-testid="consent-type-service-2-logo"
                  src="https://cdn-dev-playground.cariva.co.th/terms/images/logo.png"
                />
              </div>
              <div
                class="MuiStack-root emotion-16"
              >
                <span
                  class="MuiTypography-root MuiTypography-bodyMedium emotion-17"
                  data-testid="consent-type-service-2-title"
                >
                  ASR
                </span>
                <svg
                  data-testid="consent-type-service-2-detail-icon"
                />
              </div>
              <div
                class="MuiStack-root emotion-18"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-19"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    class="emotion-20"
                  />
                  <span
                    class="MuiTouchRipple-root emotion-21"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-22"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper MuiCollapse-vertical emotion-23"
              >
                <div
                  class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-24"
                >
                  <div
                    class="MuiStack-root emotion-25"
                    data-testid="consent-type-service-2-content"
                  >
                    <div
                      class="emotion-9"
                      data-testid="consent-type-content"
                    >
                      **Service Overview**

ASR (Automatic Speech Recognition) converts spoken language into text...

**Features**

- High accuracy speech-to-text
- Supports multiple languages
- Real-time processing

**Usage Instructions**

1. Select your language.
2. Speak into the microphone.
3. View the transcribed text.
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiStack-root emotion-12"
          >
            <div
              class="MuiStack-root emotion-13"
              data-testid="consent-type-service-3"
            >
              <div
                class="MuiBox-root emotion-14"
              >
                <img
                  alt=""
                  class="emotion-15"
                  data-testid="consent-type-service-3-logo"
                  src="https://cdn-dev-playground.cariva.co.th/terms/images/logo.png"
                />
              </div>
              <div
                class="MuiStack-root emotion-16"
              >
                <span
                  class="MuiTypography-root MuiTypography-bodyMedium emotion-17"
                  data-testid="consent-type-service-3-title"
                >
                  Symptom Checker
                </span>
                <svg
                  data-testid="consent-type-service-3-detail-icon"
                />
              </div>
              <div
                class="MuiStack-root emotion-18"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-19"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    class="emotion-20"
                  />
                  <span
                    class="MuiTouchRipple-root emotion-21"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-22"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper MuiCollapse-vertical emotion-23"
              >
                <div
                  class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-24"
                >
                  <div
                    class="MuiStack-root emotion-25"
                    data-testid="consent-type-service-3-content"
                  >
                    <div
                      class="emotion-9"
                      data-testid="consent-type-content"
                    >
                      **Service Overview**

Symptom Checker helps you understand potential causes for your symptoms...

**Features**

- Symptom input
- Potential diagnoses
- Suggested next steps

**Usage Instructions**

1. Enter your symptoms.
2. View the potential diagnoses.
3. Follow the suggested next steps for further action.
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <form
        class="emotion-72"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd emotion-73"
          data-testid="agreement-checkbox"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium emotion-74"
            data-testid="agreement-checkbox-item-button"
          >
            <input
              class="PrivateSwitchBase-input emotion-75"
              data-indeterminate="false"
              data-testid="agreement-checkbox-value"
              name="agreement"
              type="checkbox"
              value="false"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-76"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root emotion-21"
            />
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-78"
            data-testid="agreement-checkbox-label"
          >
            ฉันยอมรับเงื่อนไขและข้อตกลงการให้บริการ
          </p>
        </label>
        <div
          class="MuiStack-root emotion-79"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary emotion-80"
            data-testid="submit-button"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            ถัดไป
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
